### PR TITLE
Fixes Cannot read property 'toJSON' of null

### LINKED ISSUE #2132: GraphQL Bookshelf error with `toJSON` on null values
**Informations**
- **Node.js version**: v10
- **npm version**: v6.1
- **Strapi version**: 3.0.0-alpha.14.3
- **Database**: MariaDB 10.2
- **Operating system**: Ubuntu 18.04

**What is the current behavior?**

It seems GraphQL is having an issue in bookshelf if a value is null in a deeply nested query, this results in the response including improper or repeating data.

GraphQL Query Log Errors:

```
  "errors": [
    {
      "message": "Cannot read property 'toJSON' of null",
      "locations": [
        {
          "line": 40,
          "column": 9
        }
      ],
      "path": [
        "gssites",
        11,
        "activeObelisks",
        3,
        "activeObelisk",
        "gsCodexData"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "TypeError: Cannot read property 'toJSON' of null",
            "    at Function.findOne (/srv/api/capi-dev/plugins/content-manager/config/queries/bookshelf.js:187:25)",
            "    at process._tickCallback (internal/process/next_tick.js:68:7)"
          ]
        }
      }
    },
```

**Steps to reproduce the problem**

I can provide my full query and my public development server, outside of that it seems the error is pretty straight forward dealing with bookshelf support and graphQL specifically here: `plugins/content-manager/config/queries/bookshelf.js:187:25`

**What is the expected behavior?**

GraphQL shouldn't have any errors :wink: 

**Suggested solutions**

Unsure as I am not familiar with GraphQL or how it looks up data.

### PR DESCRIPTION
**Pull request by greatwitenorth**
My PR is a:

🐛 Bug fix #2132

Main update on the:

Framework

When using GraphQL with bookshelf if a value is null in a deeply nested query, it display the following error:

```
{
      "message": "Cannot read property 'toJSON' of null",
      "locations": [
        {
          "line": 5,
          "column": 5
        }
      ],
      "path": [
        "dealersWithinRange",
        0,
        "contact"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "TypeError: Cannot read property 'toJSON' of null",
            "    at Function.findOne (/myproject/plugins/content-manager/config/queries/bookshelf.js:187:25)"
          ]
        }
      }
    },
```

This PR checks to see if record is null and makes sure the toJSON method exists.

Related to:
https://github.com/strapi/strapi/issues/2132

### CODE REVIEW

File: packages/strapi-plugin-content-manager/config/queries/bookshelf.js [hunk 1]
```diff
@@ -184,7 +184,7 @@ module.exports = {
         withRelated: populate || this.associations.map(x => x.alias)
       });
 
-    const data = record.toJSON ? record.toJSON() : record;
+    const data = record && ('toJSON' in record) ? record.toJSON() : record;
```
[the remark this answers is not in the document]
Comment (greatwitenorth) [reply packages/strapi-plugin-content-manager/config/queries/bookshelf.js]: I don't have much experience with lodash, but it looks like that should work as long as lodash executes the function. I'm not at my dev machine at the moment, so I'll report back the exact version of node. But I'm positive it was using 10.x

[the remark this answers is not in the document]
Comment (lauriejim) [reply packages/strapi-plugin-content-manager/config/queries/bookshelf.js]: I apply @maturanomx feedback.
This PR is fixed https://github.com/strapi/strapi/issues/2132.
I think @maturanomx you speak about an other issue. Can you please create or link an issue about that.
Thank you.

[the remark this answers is not in the document]
Comment (maturanomx) [reply packages/strapi-plugin-content-manager/config/queries/bookshelf.js]: > I think @maturanomx you speak about an other issue. Can you please create or link an issue about that.

🤔 ... I never mention other issue. I just talked about the modified code of this PR. Sorry if I was not clear.

But, good to know the suggestions works 👍

**Comment by merijnvanessen** [comment]
GraphQL error: Cannot read property 'toJSON' of null
When there is no breed added to the puppy this error pops up

My dependencies
`  "dependencies": {
    "knex": "latest",
    "lodash": "4.x.x",
    "mysql": "2.16.0",
    "strapi": "^3.0.0-alpha.14.5",
    "strapi-hook-bookshelf": "^3.0.0-alpha.14.5",
    "strapi-hook-knex": "^3.0.0-alpha.14.5"
  },`

My query

` query KennelQuery($id: ID!) {
    kennel(id: $id) {
      id
      name
      puppies {
        name
        slug
        description
        breeds {
          name
        }
        thumb {
          url
        }
      }
    }
  }`



The relation (breeds has many puppies)
schermafbeelding 2018-11-14 om 11 46 21

Full error

schermafbeelding 2018-11-14 om 11 47 10

**Comment by merijnvanessen** [comment]
After a fresh install, everything works now. Thanks!

**Pull request merged**